### PR TITLE
[RTM] Use the slug generator service for the event alias

### PR DESCRIPTION
**Pull request by ausi**
See contao/core-bundle#1016

### CODE REVIEW

**Review by aschempp — APPROVED** [review]
LGTM (appart from coding style)

File: src/Resources/contao/dca/tl_calendar_events.php [hunk 1]
```diff
@@ -670,7 +670,18 @@ public function generateAlias($varValue, DataContainer $dc)
...
+			$objPage = PageModel::findWithDetails($objCalendar->jumpTo);
+			if ($objPage)
+			{
+				$slugOptions['locale'] = $objPage->language;
+				if ($objPage->validAliasCharacters)
+				{
+					$slugOptions['validChars'] = $objPage->validAliasCharacters;
+				}
+			}
+			$varValue = System::getContainer()->get('contao.slug.generator')->generate(StringUtil::stripInsertTags($dc->activeRecord->title), $slugOptions);
```
Comment (leofeyer) [inline src/Resources/contao/dca/tl_calendar_events.php]: Why did you add `StringUtil::stripInsertTags()` here?

Comment (ausi) [reply src/Resources/contao/dca/tl_calendar_events.php]: Probably because `StringUtil::generateAlias()` also stripped the insert tags in the past.

Comment (leofeyer) [reply src/Resources/contao/dca/tl_calendar_events.php]: You are right. But `StringUtil::generateAlias()` also did `decodeEntities()` and `restoreBasicEntities()`. What about those?

Comment (ausi) [reply src/Resources/contao/dca/tl_calendar_events.php]: That is not necessary I think because `$dc->activeRecord->title` is already decoded and basic-entity-restored. In my test the title `foo < bar [nbsp] baz {{br}}` correctly generated the alias `foo-bar-baz`

**Review by leofeyer — APPROVED** [review]

File: src/Resources/contao/dca/tl_calendar_events.php [hunk 2]
```diff
@@ -670,7 +670,18 @@ public function generateAlias($varValue, DataContainer $dc)
 		if ($varValue == '')
 		{
 			$autoAlias = true;
-			$varValue = StringUtil::generateAlias($dc->activeRecord->title);
+			$slugOptions = [];
+			$objCalendar = CalendarModel::findByPk($dc->activeRecord->pid);
```
Comment (leofeyer) [inline src/Resources/contao/dca/tl_calendar_events.php]: We should handle the case `$objCalendar === null`, too.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: src/Resources/contao/dca/tl_calendar_events.php (hunk 2 above)
Comment (ausi) [reply src/Resources/contao/dca/tl_calendar_events.php]: I think you are right :)

File: src/Resources/contao/dca/tl_calendar_events.php [hunk 3]
```diff
@@ -670,7 +670,20 @@ public function generateAlias($varValue, DataContainer $dc)
...
+			if (($objCalendar = CalendarModel::findByPk($dc->activeRecord->pid)) !== null && ($objPage = PageModel::findWithDetails($objCalendar->jumpTo)) !== null)
+			{
+				$slugOptions['locale'] = $objPage->language;
+
+				if ($objPage->validAliasCharacters)
+				{
+					$slugOptions['validChars'] = $objPage->validAliasCharacters;
+				}
+
+				$varValue = System::getContainer()->get('contao.slug.generator')->generate(StringUtil::stripInsertTags($dc->activeRecord->title), $slugOptions);
```
Comment (ausi) [inline src/Resources/contao/dca/tl_calendar_events.php]: @leofeyer I think this is now nested one level too deep. Should get moved one line down.

**Comment by leofeyer** [comment]
Very good job @ausi. Thanks a bunch.
